source/extensions/common/dynamic_forward_proxy/dns_cache.h

```diff
@@ -148,6 +164,11 @@ class DnsCache {
    * @return all hosts currently stored in the cache.
    */
   virtual absl::flat_hash_map<std::string, DnsHostInfoSharedPtr> hosts() PURE;
+
+  /**
+   * @return A pointer to resource manager for dns cache.
+   */
+  virtual absl::optional<DnsCacheResourceManagerPtr> dnsCacheResourceManager() PURE;
```

Comment:
do you want to differentiate `nullopt` vs` nullptr` vs `some pointer`? If no, the return type should be just a shared ptr.